src/components/CodeInput/__test__/codeInput.spec.js

```diff
@@ -0,0 +1,159 @@
+                .prop('isActive'),
+        ).toBe(true);
+    });
+    it('should keep isActive as true on the last input when all inputs are filled an last input changes', () => {
+        const component = mount(<CodeInput />);
+        useValueState.mockImplementation(() => ['1', '2', '3', '4']);
+        useFocusedIndexState.mockImplementation(() => 3);
+        component.setProps({ value: '1234' });
+        expect(
+            component
+                .find(InputItem)
+                .at(3)
+                .prop('isActive'),
+        ).toBe(true);
+    });
+    it('should call onChange with the number value when input value changes and is number', () => {
+        getNormalizedValue.mockImplementation(() => '3');
+        const onChangeFn = jest.fn();
+        const component = mount(<CodeInput onChange={onChangeFn} />);
+        const inputIndex = 0;
+        component
+            .find(InputItems)
+            .props()
+            .onChange('3', inputIndex);
+        expect(onChangeFn).toHaveBeenCalledWith('3');
+    });
+    it('should call onChange with empty string when input value changes and is not a number', () => {
+        getNormalizedValue.mockImplementation(() => '');
+        const onChangeFn = jest.fn();
+        const component = mount(<CodeInput onChange={onChangeFn} />);
+        const inputIndex = 0;
+        component
+            .find(InputItems)
+            .props()
+            .onChange('A', inputIndex);
+        expect(onChangeFn).toHaveBeenCalledWith('');
+    });
+    it('should call getNumbersFromClipboard with the copy and pasted value', () => {
+        getNumbersFromClipboard.mockClear();
+        const component = mount(<CodeInput />);
+        component
+            .find(InputItem)
+            .at(0)
+            .props()
+            .onPaste({
+                clipboardData: {
+                    getData: () => '12345abcdfe',
+                },
+            });
+        expect(getNumbersFromClipboard).toHaveBeenCalledWith('12345abcdfe');
+    });
+    it('should call getNormalizedValue with the right values when value changes', () => {
+        useValueState.mockImplementation(() => ['1', '2', '', '']);
+        const component = mount(<CodeInput value="12" />);
+        const inputIndex = 2;
+        component
+            .find(InputItems)
+            .props()
+            .onChange('3', inputIndex);
+        expect(getNormalizedValue).toHaveBeenCalledWith('3', 2, ['1', '2', '', '']);
+    });
+    it('should call useValueState with the right value and length', () => {
+        useValueState.mockImplementation(() => ['1', '', '', '']);
+        mount(<CodeInput value="1" />);
+        expect(useValueState).toHaveBeenCalledWith('1', 4);
+    });
+    it('should call useFocusedIndexState with the right value and length', () => {
+        useFocusedIndexState.mockImplementation(() => 1);
+        mount(<CodeInput value="1" />);
+        expect(useValueState).toHaveBeenCalledWith('1', 4);
```

Comment:
when testing with what was called a mocked function you need to call mockReset before since in the last test useValueState  was called with this same value then this is not testing really that now it was called with ```'1', 4```